### ISSUE TITLE
Error when using Vercel and getPrices function

### ISSUE THREAD

**Issue by dmanc** [spiceai/spice.js#68]
```
Uncaught Exception 	{"errorType":"TypeError","errorMessage":"Invalid URL","code":"ERR_INVALID_URL","input":"///v0.1/prices/eth","stack":["TypeError [ERR_INVALID_URL]: Invalid URL","    at new NodeError (node:internal/errors:399:5)","    at new URL (node:internal/url:560:13)","    at ClientRequest.<anonymous> (/var/task/___vc/chunk-MOXFNKOD.js:135:40)","    at Object.onceWrapper (node:events:628:26)","    at ClientRequest.emit (node:events:525:35)","    at HTTPParser.parserOnIncomingClient [as onIncoming] (node:_http_client:701:27)","    at HTTPParser.parserOnHeadersComplete (node:_http_common:119:17)","    at TLSSocket.socketOnData (node:_http_client:542:22)","    at TLSSocket.emit (node:events:513:28)","    at addChunk (node:internal/streams/readable:324:12)"]}
Unknown application error occurred
Runtime.Unknown
```

We think it's related to this line of code: https://github.com/spiceai/spice.js/blob/46fa578063c28426d9a1078260b08a2d5e6c53e3/src/client.ts#LL100C7-L100C7

Could it be because vercel has their own implementation of fetch? https://github.com/vercel/fetch

**Comment by ewgenius** [comment]
Hello, could you please provide more details? Is that next.js project, or just serverless function using vercel runtime? Which version of node.js you have?

**Comment by dmanc** [comment]
Yeah it's a next.js project, node version is 18.16.18 and the error log comes from a serverless function. We are trying the latest version of this template: https://github.com/t3-oss/create-t3-app/tree/next without nextauth.js. 

Here is the price router under `src/server/api/routers/price.ts`:
```
import { z } from "zod";
import { createTRPCRouter, publicProcedure } from "~/server/api/trpc";
import { env } from "~/env.mjs";
import { SpiceClient } from "@spiceai/spice";
import type { LatestPrice } from "@spiceai/spice/dist/interfaces";

export const priceRouter = createTRPCRouter({
  getPriceData: publicProcedure
    .input(z.object({ token: z.string() }))
    .query(async ({ input }) => {
        return {
            avg: `Average price of ${input.token} is: ` + (await fetchPriceData(input.token)).avePrice
        }
    })
});

export async function fetchPriceData(token: string): Promise<LatestPrice> {
    const API_KEY = env.GOERLI_APP_SPICE_API_KEY
    const spiceClient = new SpiceClient(API_KEY, 'data.spiceai.io:443');
    const price = spiceClient.getPrice(token)

    return price
}
```

**Comment by ewgenius** [comment]
Thanks for details! It was caused by extra slashes in api endpoint, fixed in #69 
@dmanc please update to `0.3.3`, that should resolve the problem

**Comment by dmanc** [comment]
Thanks just updated and everything works.
